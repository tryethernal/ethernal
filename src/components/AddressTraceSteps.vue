<template>
    <Trace-Steps-Table
        :loading="loading"
        :items="items"
        :total-items="totalItems"
        :dense="dense"
        :highlight-address="address"
        @update:options="fetchAddressTraceSteps" />
</template>

<script setup>
import { ref } from 'vue';
import { inject } from 'vue';
import TraceStepsTable from './TraceStepsTable.vue';

const props = defineProps({
    address: {
        type: String,
        required: true
    },
    dense: Boolean,
});

// State
const loading = ref(false);
const items = ref([]);
const totalItems = ref(0);

const $server = inject('$server');

const fetchAddressTraceSteps = async ({ page, itemsPerPage }) => {
    loading.value = true;
    try {
        const { data } = await $server.getAddressInternalTransactions(props.address, page, itemsPerPage);
        items.value = data.items;
        totalItems.value = items.value.length == itemsPerPage ?
            (page * itemsPerPage) + 1 :
            page * itemsPerPage;
    } catch (error) {
        console.error('Error fetching address trace steps:', error);
    } finally {
        loading.value = false;
    }
};
</script> 