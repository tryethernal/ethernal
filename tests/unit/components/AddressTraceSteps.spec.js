import flushPromises from 'flush-promises'

import AddressTraceSteps from '@/components/AddressTraceSteps.vue';

const stubs = ['TraceStepsTable'];

describe('AddressTraceSteps.vue', () => {
    it('Should load address trace steps', async () => {
        vi.spyOn(server, 'getAddressInternalTransactions').mockResolvedValueOnce({
            data: {
                items: [
                    {
                        transactionHash: '0x1234567890123456789012345678901234567890',
                        blockNumber: 1,
                        timestamp: '2021-01-01',
                    }
                ]
            }
        });

        const wrapper = mount(AddressTraceSteps, {
            props: {
                address: '0x1234567890123456789012345678901234567890',
                dense: false
            },
            global: {
                stubs
            }
        });
        wrapper.vm.fetchAddressTraceSteps({ page: 1, itemsPerPage: 10 });
        await flushPromises();

        expect(wrapper.html()).toMatchSnapshot();
    });
});
