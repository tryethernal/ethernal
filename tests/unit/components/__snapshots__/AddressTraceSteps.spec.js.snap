// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`AddressTraceSteps.vue > Should load address trace steps 1`] = `"<trace-steps-table-stub items="[object Object]" totalitems="10" dense="false" highlightaddress="0x1234567890123456789012345678901234567890" loading="false" sortby="[object Object]"></trace-steps-table-stub>"`;
